pkg/util/nsenter/nsenter.go

```diff
@@ -79,24 +80,29 @@ func NewNsenter() *Nsenter {
 			"ls":          "",
 			"sh":          "",
 			"chmod":       "",
+			"nsenter":     "",
 		},
 	}
 	// search for the required commands in other locations besides /usr/bin
+	searchPaths := []string{"/bin", "/usr/sbin", "/usr/bin", "/"}
 	for binary := range ne.paths {
-		// default to root
-		ne.paths[binary] = filepath.Join("/", binary)
-		for _, path := range []string{"/bin", "/usr/sbin", "/usr/bin"} {
+		found := false
+		for _, path := range searchPaths {
 			binPath := filepath.Join(path, binary)
 			if _, err := os.Stat(filepath.Join(hostRootFsPath, binPath)); err != nil {
 				continue
 			}
 			ne.paths[binary] = binPath
+			found = true
 			break
 		}
-		// TODO: error, so that the kubelet can stop if the paths don't exist
-		// (don't forget that systemd-run is optional)
+
+		// systemd-run is optional
+		if binary != "systemd-run" && !found {
```

Comment:
This does not seem like it belongs in this method.  Why are you checking for optional binaries inside a generic function, instead of the place that calls systemd-run?

Reply:
@smarterclayton why is this bad? That check is only making the binary being optional. Unless you do this in the constructor you end up hacking it with checking every time if it was already searched/initialized and you'll do the path searching in 2 places.

Reply:
@tnozicka @smarterclayton Updated. PTAL. Thanks.